spring-boot-project/spring-boot-docs/src/docs/asciidoc/using/devtools.adoc

```diff
@@ -217,10 +217,18 @@ For IntelliJ IDEA, you can follow the https://www.jetbrains.com/help/idea/spring
 As described earlier in the <<using#using.devtools.restart.restart-vs-reload>> section, restart functionality is implemented by using two classloaders.
 For most applications, this approach works well.
 However, it can sometimes cause classloading issues.
+These might manifest in an exception like:
+
+[source,text]
+----
+Caused by: java.lang.IllegalArgumentException: non-public interface is not defined by the given loader
+----
```

Comment:
I'm not in favour of specifically mentioning `IllegalArgumentException` here. Unfortunately, class loading issues can appear in many different ways. Mentioning one specifically is likely to do more harm than good.

Reply:
It's less about the `IllegalArgumentException` and more about the error message itself. Recently, I was called in to troubleshoot exactly this problem. At first I was stumped because the application ran fine in the IDE and when packaged. It took me a lot of time to figure out what the problem was. Feeding the error message to search engines didn't yield good results, either. 

Do you have an idea how to make this easier to discover without misleading people that have other classloading problems?

Reply:
I'm afraid not. It's a shame that the JDK uses "loader" rather than the arguably less ambiguous "class loader" as that may help to point people in the right direction.

The docs on DevTools currently mention "classloading issues" but don't really expand on how to  diagnose them. Rather than trying to list specific failures that may be a classloading issue, I think it would be better to point people towards [completely disabling restart](https://docs.spring.io/spring-boot/docs/2.5.x/reference/htmlsingle/#using.devtools.restart.disable) as a way of identifying if the split class loader is the cause.